src/toil_scripts/gatk_germline/germline.py

```diff
@@ -389,50 +335,187 @@ def gatk_haplotype_caller(job, bam_id, bai_id, config, emit_threshold=10.0, call
+        return
+
+    # VQSR requires many variants in order to train a decent model. GATK recommends a minimum of
+    # 30 exomes or one large WGS sample:
+    # https://software.broadinstitute.org/gatk/documentation/article?id=3225
+    filtered_vcfs = {}
+    if config.run_vqsr or len(gvcfs) > 30:
+        if config.joint:
+            joint_vcf = job.addFollowOnJobFn(vqsr_pipeline, gvcfs, config)
+            filtered_vcfs = joint_vcf.addChildJobFn(split_vcf_by_name, gvcfs.keys(),
+                                                    joint_vcf.rv(), config).rv()
+        else:
+            for uuid, gvcf in gvcfs.iteritems():
+                filtered_vcfs[uuid] = job.addFollowOnJobFn(vqsr_pipeline,
+                                                           dict(uuid=gvcf),
+                                                           config).rv()
+    else:
+        for uuid, gvcf in gvcfs.iteritems():
+            filtered_vcfs[uuid] = job.addFollowOnJobFn(hard_filter_pipeline,
+                                                       uuid, gvcf,
+                                                       config).rv()
+    return filtered_vcfs
+
+
+def gatk_germline_pipeline(job, uuid, url, config, url2=None, rg_line=None):
+    """
+    Runs GATK Germline Pipeline on a single sample. Writes gvcf to output directory
+    defined in the config dictionary. Removes secondary alignments.
+
+    0: Align Input or Download BAM file
+    1: Generate BAM Index
+    2: GATK Preprocessing (Optional)
+    3: GATK HaplotypeCaller
+    4: Write GVCF to output directory
+
+    :param JobFunctionWrappingJob job: passed automatically by Toil
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to sample bam or fq file
+    :param Namespace config: Configuration options for pipeline
+    :param str url2: Paired FASTQ URL. Default is None.
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: GVCF FileStoreID
+    :rtype: str
+    """
+    config = deepcopy(config)
+    config.uuid = uuid
+    config.url = url
+
+    get_bam = job.wrapJobFn(prepare_bam, uuid, url, url2, config, rg_line=rg_line).encapsulate()
+    job.addChild(get_bam)
+    input_bam = get_bam.rv(0)
+    input_bai = get_bam.rv(1)
+
+    if config.preprocess:
+        preprocess = job.wrapJobFn(run_gatk_preprocessing,
+                                   config.cores,
+                                   get_bam.rv(0),
+                                   get_bam.rv(1),
+                                   config.genome_fasta,
+                                   config.genome_dict,
+                                   config.genome_fai,
+                                   config.phase,
+                                   config.mills,
+                                   config.dbsnp,
+                                   xmx=config.xmx).encapsulate()
+        get_bam.addChild(preprocess)
+
+        if config.preprocess_only:
```

Comment:
I really don't like putting this logic here, because now the function can either return `(bam ID, bai ID)` or `gvcf ID` and that seems like a bad smell.

Also, all of these forks need inline documentation.